Sources/LiveKit/Core/Room+Region.swift

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 LiveKit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+// MARK: - Room+Region
+
+extension Room {
+    static let defaultCacheInterval: TimeInterval = 3000
+
+    func resolveBestRegion() async throws -> RegionInfo {
+        try await requestRegionSettings()
+
+        let sortedByDistance = _regionState.remaining.sorted { $0.distance < $1.distance }
```

Comment:
I think we should trust the server to have sorted the regions already. there might be other reasons for server ordering rather than just distance